workflow/train_test_split.py

```diff
@@ -0,0 +1,165 @@
+from collections.abc import Iterable
+import numpy as np
+import pandas as pd
+import pytest
+
+
+def train_test_split(df, key, test_size=0.2, random_state=None, stratify=None):
+
+    original_random_state = np.random.get_state()
+    np.random.seed(random_state)
+
+    if stratify is None:
+        test_df = get_test_split(df, key, test_size)
+    else:
+        test_df = multilabel_stratified_split(df, key, stratify, test_size)
+
+    train_df = df[~df[key].isin(test_df[key])]
+    np.random.set_state(original_random_state)
+
+    return train_df, test_df
+
+
+def get_test_split(df, key, test_size, resolution=100000):
+    salt = np.random.randint(0, resolution)
+    split_key = df[key].apply(lambda x: ((hash(hash(x) + salt) % resolution)))
+    return df[split_key < resolution * test_size]
+
+
+def multilabel_stratified_split(df, key, stratify, test_size):
```

Comment:
This looks like the implementation that I talked about before but it should be weaker than the old one? Am I missing something?